libkbfs/block_retrieval_worker.go

```diff
@@ -48,12 +48,12 @@ func newBlockRetrievalWorker(bg blockGetter,
 // blockGetter.getBlock, and responds to the subscribed requestors with the
 // results.
 func (brw *blockRetrievalWorker) HandleRequest() (err error) {
-	brw.queue.Work(brw.workCh)
 	var retrieval *blockRetrieval
 	select {
-	case retrieval = <-brw.workCh:
+	case <-brw.workCh:
+		retrieval = brw.queue.popIfNotEmpty()
```

Comment:
OK I agree with the logic, but I don't agree that it can realistically happen. It'd require the prefetching workers to be successfully completing every request and none of the on-demand workers to be. On average I suspect that this would be rare, with a small percentage of prefetch requests addressed by on-demand workers.

I could instrument it to see how often it happens though.